### ISSUE TITLE
testing: reported failure with go test -json

### ISSUE THREAD

**Issue by eliben** [google/go-cloud#2041]
In a clean clone, running `go test ./...` doesn't report failures. 

However, running `go test -json ./...` reports one failing test:

```
{"Time":"2019-05-09T15:26:51.338409101-07:00","Action":"fail","Package":"gocloud.dev/tests/aws/app","Elapsed":0.101}
```

Even though the return code from `go test` is still 0.

**Comment by eliben** [comment]
Opened https://github.com/golang/go/issues/31969

**Comment by eliben** [comment]
For us this is fixed by #2049 
Upstream bug tracking underlying fix
